app/controllers/public/addresses_controller.rb

```diff
@@ -1,16 +1,38 @@
 class Public::AddressesController < ApplicationController
   def index
+    @address = Address.new
+    @addresses = Address.where(customer: current_customer)
   end
 
   def create
+    @address = Address.new(address_params)
+    @address.customer = current_customer
+    if @address.save
+      redirect_to addresses_path
+    else
+      render :index
+    end
   end
 
   def edit
+    @address = Address.find(params[:id])
   end
 
   def update
+    @address = Address.find(params[:id])
+    @address.update(address_params)
+    redirect_to addresses_path
```

Comment:
更新処理に失敗した場合は再度編集ページを表示するようにしましょう
```suggestion
    if @address.update(address_params)
      redirect_to addresses_path
    else
      render :edit
    end
```